src/coprocessor/dag/expr/builtin_string.rs

```diff
@@ -37,6 +37,30 @@ impl ScalarFunc {
         let i = try_opt!(self.children[0].eval_int(ctx, row));
         Ok(Some(Cow::Owned(format!("{:b}", i).into_bytes())))
     }
+
+    pub fn left<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let s = try_opt!(self.children[0].eval_string(ctx, row));
+        let i = try_opt!(self.children[1].eval_int(ctx, row));
+        let s = str::from_utf8(&s)?;
+        let slen = s.chars().count();
+        let l: usize = if i < 0 {
+            0
+        } else if i as usize > slen {
+            slen
```

Comment:
Could we return empty or the whole string for `i<0` and `i>slen` directly?

Reply:
Reasonable. I'll fix here